File: yjj.py
import loguru
import requests
import re
import execjs
headers = {
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7",
    "Accept-Language": "zh-CN,zh;q=0.9",
    "Cache-Control": "no-cache",
    "Connection": "keep-alive",
    "Pragma": "no-cache",
    "Sec-Fetch-Dest": "document",
    "Sec-Fetch-Mode": "navigate",
    "Sec-Fetch-Site": "none",
    "Sec-Fetch-User": "?1",
    "Upgrade-Insecure-Requests": "1",
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36",
    "sec-ch-ua": "\"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Google Chrome\";v=\"114\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\""
}
url = "https://www.nmpa.gov.cn/datasearch/home-index.html"
response = requests.get(url, headers=headers)
NfBCSins2OywO = response.cookies.get('NfBCSins2OywO')
acw_tc = response.cookies.get('acw_tc')
# print(response.content.decode('utf-8'))
arg1 = re.findall('content="(.*?)" r="',response.text)[0]
arg2 = re.findall("t\" r='m'>(.*?)</script><script ty",response.text)[0]
# print(arg1, arg2)
jscode = open('./yjj.js',encoding='utf-8').read().replace('arg1_content',arg1).replace('"arg2_js"',arg2)
NfBCSins2OywP = execjs.compile(jscode).call('get_cookie')
print(NfBCSins2OywP)
wp = NfBCSins2OywP.split('wP=')[1].split('; path=/')[0]
cookies = {
    "acw_tc": acw_tc,
    "NfBCSins2OywO": NfBCSins2OywO,
    "NfBCSins2OywP": wp,
    "STEP_TIPS_INDEX": "true",
    "token": "",
    "STEP_TIPS_RESULT": "true",
}
url = "https://www.nmpa.gov.cn/datasearch/home-index.html"
res = requests.get(url, headers=headers, cookies=cookies)
loguru.logger.info(res.content.decode('utf-8'))

File: yjj.js
window = global
proxy_array = ['window','document','navigator','location']
var content = "arg1_content";
null_function = function (){console.log(arguments)}
delete __filename;
delete __dirname;


null_function = function (){console.log(arguments)}

window = global
window.top = window
window.setInterval = null_function
window.ActiveXObject = undefined
window.addEventListener = null_function

document = {
    characterSet:'UTF-8',
    charset:'UTF-8',
    scripts: ['script','script']
}
document.removeChild = null_function
div = {getElementsByTagName(val){
    if(val==='i'){ return {length :0}}
    }}
document.createElement = function (val){if(val==='div'){return div} if (val === 'form'){
        return {}
    } if (val === 'a'){
        return {}
    }}
document.getElementsByTagName = function (val){
        return [
            {
                content:content,
                parentNode:{
                    removeChild:function (){}
                },
                parentElement : {
                     removeChild:function (){}
                }
            },
            {
                 content:content,
                 parentNode:{
                    removeChild:function (){}
                },
                parentElement : {
                     removeChild:function (){}
                }
            }
        ]
}
document.getElementById = null_function

Object.prototype.getAttribute = function (val){if(val==='r'){return 'm'}}
location = {
    "ancestorOrigins": {},
    "href": "https://www.nmpa.gov.cn/datasearch/home-index.html",
    "origin": "https://www.nmpa.gov.cn",
    "protocol": "https:",
    "host": "www.nmpa.gov.cn",
    "hostname": "www.nmpa.gov.cn",
    "port": "",
    "pathname": "/datasearch/home-index.html",
    "search": "",
    "hash": ""
}
navigator = {
    userAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36",
    languages: ["zh-CN", "zh"],
    appVersion: "5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36",
    webdriver: false,
    appName: "Netscape",
    vendor: "Google Inc.",
    connection:  {
        downlink: 10,
        effectiveType: "4g",
        rtt: 200,
        saveData: false,
    }
}


"arg2_js";
require('./15QqdX9S7nDE')


function get_cookie(){
    return document.cookie
}
console.log(document.cookie);